modules/database/src/ioc/db/dbAccessNoThrow.cpp

```diff
@@ -0,0 +1,181 @@
+/*************************************************************************\
+* Copyright (c) 2021 Lucas Russo
+* SPDX-License-Identifier: EPICS
+* EPICS BASE is distributed subject to a Software License Agreement found
+* in file LICENSE that is included with this distribution.
+\*************************************************************************/
+
+#include <stdexcept>
+
+#include "recSup.h"
+#include "dbAccessDefs.h"
+#include "errlog.h"
+#include "dbAccessNoThrow.h"
+
+/*
+ * Maps possible excpetions to DB error codes. Using the
+ * "exception dispatcher" pattern.
+ */
+inline long dbExceptionToLong()
+{
+    try {
+        throw;
+    }
+    catch (std::invalid_argument& e) {
+        errlogPrintf("dbExceptionToLong: %s\n", e.what());
+        return S_db_errArg;
+    }
+    catch (std::bad_alloc& e) {
+        errlogPrintf("dbExceptionToLong: %s\n", e.what());
+        return S_db_noMemory;
+    }
+    catch (std::exception& e) {
+        errlogPrintf("dbExceptionToLong: Unhandled exception\n");
+        /* Return something more meaningful? */
+        return S_db_errArg;
+    }
+}
+
+/*
+ * Pseudo variadic templates for use without C++11 features
+ */
+template <typename F>
+long dbRecSupNoThrowWrapper(F f) {
+    try {
+        return f();
+    }
+    catch(...){
+        return dbExceptionToLong();
+    }
+}
+
+template <typename F, typename T1>
+long dbRecSupNoThrowWrapper(F f, T1 t1) {
+    try {
+        return f(t1);
+    }
+    catch(...){
+        return dbExceptionToLong();
+    }
+}
+
+template <typename F, typename T1, typename T2>
+long dbRecSupNoThrowWrapper(F f, T1 t1, T2 t2) {
+    try {
+        return f(t1, t2);
+    }
+    catch(...){
+        return dbExceptionToLong();
+    }
+}
+
+template <typename F, typename T1, typename T2, typename T3>
+long dbRecSupNoThrowWrapper(F f, T1 t1, T2 t2, T3 t3) {
+    try {
+        return f(t1, t2, t3);
+    }
+    catch(...){
+        return dbExceptionToLong();
+    }
+}
+
+/*
+ * Wraps all RecordSupport functions from throwable functions to no-throwable ones,
+ * without using C++11 features
```

Comment:
Well, this does give a good example of why I didn't do this initially.  For the moment, let's only add wrappers as they used.

Reply:
Ok, sounds good.